dogfood.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
```

Comment:
As a convention, we put all the tooling inside `tools/<tool-name>/*` and add a `make` command to run it. We can start by moving this script to `tools/dogfooding/dogfood.py` in this PR and then we will add `make dogfooding-pr` command when setting up the CI to run it 🙌.